### ISSUE TITLE
the -x option disables SC2039 but it should not

### ISSUE THREAD

**Issue by pabs3** [koalaman/shellcheck#1181]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2039
- My shellcheck version (`shellcheck --version` or "online"): 0.4.6

#### Here's a snippet or screenshot that shows the problem:

```sh
$ head foo.sh saywhat.sh 
==> foo.sh <==
#!/bin/sh
source ./saywhat.sh

==> saywhat.sh <==
echo output from saywhat.sh
$ bash foo.sh
output from saywhat.sh
$ dash foo.sh
foo.sh: 2: foo.sh: source: not found
```

#### Here's what shellcheck currently says:

```
$ shellcheck -x foo.sh
$ shellcheck foo.sh

In foo.sh line 2:
source ./saywhat.sh
^-- SC2039: In POSIX sh, 'source' in place of '.' is undefined.
^-- SC1091: Not following: ./saywhat.sh was not specified as input (see shellcheck -x).
```

#### Here's what I wanted or expected to see:

```
$ shellcheck -x foo.sh

In foo.sh line 2:
source ./saywhat.sh
^-- SC2039: In POSIX sh, 'source' in place of '.' is undefined.
$ shellcheck foo.sh

In foo.sh line 2:
source ./saywhat.sh
^-- SC2039: In POSIX sh, 'source' in place of '.' is undefined.
^-- SC1091: Not following: ./saywhat.sh was not specified as input (see shellcheck -x).
```

**Comment by ngzhian** [comment]
I'm still trying to understand what is happening, but I found that https://github.com/koalaman/shellcheck/blob/master/src/ShellCheck/Parser.hs#L1919 is problematic. If you create `T_Include` with the id from T_SimpleCommand, it works. I have no idea why, maybe during the output it filters some stuff?

**Comment by koalaman** [comment]
This is because ShellCheck normally doesn't output warning for included files without `-a`.

The problem here is that the AST ends up being `T_Include _ (source command) (included script)` so that the `source` command is considered part of the included script instead of the original script.

One possible fix would be to add a second node type, e.g. `T_SourceCommand _ (source command) (T_Include $ originalScript)` so that the source command is no longer under the `T_Include`

**Comment by koalaman** [comment]
Fixed by @ngzhian. Thanks!
